ui-discover/src/main/java/app/tivi/home/discover/Discover.kt

```diff
@@ -0,0 +1,424 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.preferredHeight
+import androidx.compose.foundation.layout.preferredSize
+import androidx.compose.foundation.layout.preferredWidth
+import androidx.compose.foundation.lazy.ExperimentalLazyDsl
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.text.FirstBaseline
+import androidx.compose.material.EmphasisAmbient
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.ProvideEmphasis
+import androidx.compose.material.Surface
+import androidx.compose.material.TextButton
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Refresh
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Providers
+import androidx.compose.runtime.Recomposer
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.DensityAmbient
+import androidx.compose.ui.platform.setContent
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.LiveData
+import androidx.ui.tooling.preview.Preview
+import app.tivi.common.compose.AutoSizedCircularProgressIndicator
+import app.tivi.common.compose.Carousel
+import app.tivi.common.compose.IconResource
+import app.tivi.common.compose.LogCompositions
+import app.tivi.common.compose.PosterCard
+import app.tivi.common.compose.ProvideDisplayInsets
+import app.tivi.common.compose.TiviDateFormatterAmbient
+import app.tivi.common.compose.onSizeChanged
+import app.tivi.common.compose.rememberMutableState
+import app.tivi.common.compose.statusBarsPadding
+import app.tivi.data.entities.Episode
+import app.tivi.data.entities.Season
+import app.tivi.data.entities.TiviShow
+import app.tivi.data.entities.TmdbImageEntity
+import app.tivi.data.entities.TraktUser
+import app.tivi.data.resultentities.EntryWithShow
+import app.tivi.trakt.TraktAuthState
+import app.tivi.util.TiviDateFormatter
+import com.google.android.material.composethemeadapter.MdcTheme
+import dev.chrisbanes.accompanist.coil.CoilImage
+
+internal fun ViewGroup.composeDiscover(
+    state: LiveData<DiscoverViewState>,
+    actioner: (DiscoverAction) -> Unit,
+    tiviDateFormatter: TiviDateFormatter,
+    textCreator: DiscoverTextCreator
+): Any = setContent(Recomposer.current()) {
+    Providers(
+        TiviDateFormatterAmbient provides tiviDateFormatter,
+        DiscoverTextCreatorAmbient provides textCreator
+    ) {
+        MdcTheme {
+            LogCompositions("MdcTheme")
```

Comment:
Remove?

Reply:
I like to keep these. They get removed anyway in the release build.